### ISSUE TITLE
Projections UI doesn't work with HTTPS

### ISSUE THREAD

**Issue by pmazurek** [EventStore/EventStore.UI#67]
I am setting up a evenstore cluster with SSL termination and load balancer on AWS.
My domain is pointing to the loadbalancer which is on 2113 on HTTPS, and LB is redirecting to the instances of ES wich are on HTTP 2113.

Problem is when we try to run any projection in the UI we get this error in console log:
```
Mixed Content: The page at 'https://my_domain:2113/web/index.html#/projections/http%253A%252F%252Fmy_domain%253A2113%252Fprojection%252F%2524by_event_type' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://my_domain:2113/projection/$by_event_type/result'. This request has been blocked; the content must be served over HTTPS.
```
For whatever reason UI is trying to access HTTP. Only during starting/stopping the projections. I did not notice anything else behaving badly.
My current reasearch showed that URL is based on the value we use during login. I have checked and the authdata cookie is stored correctly (meaning: with HTTPS).

**Comment by Gutek** [comment]
Yes, but with this particular case, we are using URL returned by ES during querying for all projections. imo, as ES is running under HTTP, it gives you back links to http projections. if you will notice, url when you query to projection details - it should contain http. As i can see from your's example it does:

```
https://my_domain:2113/web/index.html#/projections/http%253A%252F%252Fmy_domain%253A2113%252Fprojection%252F%2524by_event_type
```

I will need to talk to guys about this, as from UI perspective, I can only hard-code replace for http to https in this particular situation.

**Comment by jen20** [comment]
The server is returning in correct links here.

**Comment by ianbattersby** [comment]
This is also a problem outside of HTTPS, e.g. if you are hosting multiple nodes on the same box over HTTP with different port numbers. So if I browse to node 3 on 10.0.0.1:3114 with node 1 and 2 running on 127.0.0.1:1114/2114, then the OPTION calls made when browsing http://10.0.0.1:3114 are made to 127.0.0.1:2112?!
